crypto/internal.h

```diff
@@ -893,7 +894,35 @@ static inline void CRYPTO_store_word_le(void *out, crypto_word_t v) {
 }
 
 
+enum fips_approved_algorithm_t {
+  FIPS_APPROVED_NO_STATE        = 0,
+  FIPS_APPROVED_EVP_AES_128_GCM = 1,
+  FIPS_APPROVED_EVP_AES_256_GCM = 2,
+  FIPS_APPROVED_EVP_AES_128_CTR = 3,
+  FIPS_APPROVED_EVP_AES_192_CTR = 4,
+  FIPS_APPROVED_EVP_AES_256_CTR = 5,
+  FIPS_APPROVED_EVP_AES_128_CBC = 6,
+  FIPS_APPROVED_EVP_AES_192_CBC = 7,
+  FIPS_APPROVED_EVP_AES_256_CBC = 8,
+
+  FIPS_APPROVED_ALGORITHM_MAX = 8,
+};
+
 // FIPS functions.
+#if defined(AWSLC_FIPS)
+
+struct fips_service_indicator_state {
+  uint64_t counter;
```

Comment:
We can statically assert that this field is at least a certain size in e.g. the unit tests.